pallets/avn-anchor/src/lib.rs

```diff
@@ -288,9 +336,57 @@ pub mod pallet {
 
             Ok(())
         }
+
+        #[pallet::weight(<T as pallet::Config>::WeightInfo::set_checkpoint_fee())]
+        #[pallet::call_index(6)]
+        pub fn set_checkpoint_fee(
+            origin: OriginFor<T>,
+            chain_id: ChainId,
+            new_fee: BalanceOf<T>,
+        ) -> DispatchResult {
+            ensure_root(origin)?;
+
+            CheckpointFee::<T>::insert(chain_id, new_fee);
+            Self::deposit_event(Event::CheckpointFeeUpdated { chain_id, new_fee });
+
+            Ok(())
+        }
     }
 
     impl<T: Config> Pallet<T> {
+        pub(crate) fn charge_fee(handler: T::AccountId, chain_id: ChainId) -> DispatchResult {
+            let payment_nonce = PaymentNonces::<T>::get(&handler);
+            let checkpoint_fee = Self::get_checkpoint_fee(chain_id);
+            let treasury_account_id = compute_treasury_account_id::<T>();
+
+            T::FeeHandler::pay_fee(
+                &T::Token::default(),
+                &checkpoint_fee,
+                &handler,
+                &treasury_account_id,
+            )?;
+
+            <PaymentNonces<T>>::mutate(&handler, |n| *n += 1);
+
+            Self::deposit_event(Event::CheckpointFeeCharged {
+                handler: handler.clone(),
+                fee: checkpoint_fee,
+                nonce: payment_nonce,
+                chain_id,
+            });
+
+            Ok(())
+        }
+
+        pub fn get_checkpoint_fee(chain_id: ChainId) -> BalanceOf<T> {
+            let fee = CheckpointFee::<T>::get(chain_id);
+            if fee == BalanceOf::<T>::default() {
+                T::DefaultCheckpointFee::get()
```

Comment:
Would it be more explicit if it was an option query or if we were checking for BalanceOf::<T>::zero()?

Reply:
we want to be able to set the fee to 0 right?

Reply:
Yes, IMO the correct way to handle that would be to make the storage an OptionsQuery instead of a ValueQuery item. Unless I am mistaken, `BalanceOf::<T>::default()` returns 0 which would make this use the default fee.